sdl 1/2 + wasm

humbertodias opened the pull request:
## Changes
* Added support for SDL2 while retaining SDL1 compatibility.
* Added macOS compilation script
* `Rockbot_Editor.pro` adapted for QT5 on macOS
* Added `CONFIG += debug` in `RockDroid.pro`
* Added `build-wasm.sh` on https://humbertodias.github.io/rockbot/
* Creating `sdl_layer` to abstract SDL1 and SDL2
* All compilation warnings removed

### Tested on Ubuntu 25.04

https://github.com/user-attachments/assets/0117c676-4ca3-4dd9-bd55-ca1a4e2e7f3c

### Tested on macOS Sequoia

https://github.com/user-attachments/assets/a4662662-74d2-4fe8-a1f0-5d190395e9c9

### Browser - 60fps

<img width="1118" height="917" alt="Screenshot 2025-07-11 at 11 00 02 PM" src="https://github.com/user-attachments/assets/0d384049-d9f8-427b-b89e-fe92937640a5" />

humbertodias committed:
wip

humbertodias committed:
merge ubuntu branch

humbertodias committed:
inputlib with sdl2

humbertodias committed:
wip - window

humbertodias committed:
add debug mode

humbertodias committed:
compilation ok

humbertodias committed:
add SDL_SetVideoMode

humbertodias committed:
Merge remote-tracking branch 'origin/sdl2-osx' into sdl2

humbertodias committed:
editor.pro for osx

humbertodias committed:
rm warn

humbertodias committed:
fix: set_surface_alpha

humbertodias committed:
clean up

humbertodias committed:
README

humbertodias committed:
Merge branch 'rockdroid' into sdl2

humbertodias committed:
sdl2: SDL_SetWindowTitle

humbertodias committed:
add enum PLATFORM_OSX

humbertodias committed:
clean up

humbertodias committed:
add E_PLATFORM.PLATFORM_OSX

humbertodias committed:
mesa unnecessary for osx

humbertodias committed:
warning

humbertodias committed:
osx running

humbertodias committed:
todo: linux scale not working

humbertodias committed:
remove todo

protoman commented:
Can you make the change optional through a defined macro/flag?
Like #ifdef SDL2 #include <sdl2/sdl.h> #else #incvlude <SDL/sdl.h> #endif
There is a reason I did not port to SDL2 yeat, that are most platforms like PS2, Dingoo, PocketGo and others do not have support for it. I do not want to simply kill those ports yet.
I was working in a kind of abstraction layer to make switching SDL/SDL2/whatever easy like that, but never got much further with the idea.

humbertodias committed:
unused

humbertodias committed:
unused include

humbertodias committed:
lint

humbertodias committed:
supporting sdl 1 and 2

humbertodias commented:
@protoman 
I've implemented your idea into `sdl_layer.h`
We basically use the same SDL1 signature with `SDLL` prefix

```cpp
#pragma once

#ifdef SDL2

#include <SDL2/SDL.h>
#include <SDL2/SDL_main.h>
#include <SDL2/SDL_image.h>
#include <SDL2/SDL_mixer.h>
#include <SDL2/SDL_ttf.h>
#include <SDL2/SDL_endian.h>
#include <SDL2/SDL2_rotozoom.h>

// global
extern SDL_Window *window;
extern SDL_Renderer *renderer;

// fallback sdl1
#define SDL_HWSURFACE 0x00000001
#define SDL_ASYNCBLIT 0x00000004
#define SDL_ANYFORMAT 0x10000000
#define SDL_HWPALETTE 0x20000000
#define SDL_DOUBLEBUF 0x40000000
#define SDL_FULLSCREEN 0x80000000
#define SDL_RESIZABLE 0x00000010
#define SDL_NOFRAME 0x00000020

#define SDL_SRCALPHA 0x00010000
#define SDL_SRCCOLORKEY 0x00001000

#else

#include <SDL/SDL.h>
#include <SDL/SDL_image.h>
#include <SDL/SDL_mixer.h>
#include <SDL/SDL_ttf.h>
#include <SDL/SDL_endian.h>
#include <SDL/SDL_rotozoom.h>

#endif

int SDLL_SetAlpha(SDL_Surface *surface, Uint32 flag, Uint8 alpha);

SDL_Surface *SDLL_SetVideoMode(int width, int height, int bpp, Uint32 flags);

int SDLL_SoftStretch(SDL_Surface *src, SDL_Rect *srcrect, SDL_Surface *dst, SDL_Rect *dstrect);

SDL_Surface *SDLL_DisplayFormat(SDL_Surface *surface);

void SDLL_WM_SetCaption(const char *title, const char *icon);

SDL_Surface *SDLL_DisplayFormatAlpha(SDL_Surface *surface);

void SDLL_WM_SetIcon(SDL_Surface *icon, Uint8 *mask);

int SDLL_Flip(SDL_Surface *screen);

const char* SDLL_GetKeyName(int key);
```
SDL1 remains the default.
To enable SDL2, compile with DEFINES+=SDL2

```bash
# sdl1 
./build-ubuntu.sh

# sdl2
./build-ubuntu.sh sdl2
```
https://github.com/user-attachments/assets/b9fe270d-2158-4a59-8373-696dcb19e624

humbertodias committed:
make clean before

humbertodias committed:
add build-wasm.sh

humbertodias committed:
doc

humbertodias committed:
-sSDL2_MIXER_FORMATS='["mod","wav"]'

humbertodias committed:
cleanup

humbertodias committed:
-DSDL2 for osx

humbertodias committed:
cleanup

humbertodias committed:
avoiding repetition

humbertodias committed:
abstracting SDLL_SetAlpha

humbertodias committed:
abstracting SDLL_SetVideoMode

humbertodias committed:
abstracting SDLL_SoftStretch

humbertodias committed:
abstracting SDLL_DisplayFormat

humbertodias committed:
make clean build/rockbot

humbertodias committed:
abstracting SLDL_WM_SetCaption

humbertodias committed:
typo

humbertodias committed:
typo

humbertodias committed:
abstracting SDLL_WM_SetIcon

humbertodias committed:
abstracting SDLL_GetKeyName

humbertodias committed:
hadle sdl2 SDL_WINDOWEVENT_RESIZED

humbertodias committed:
SDL_HWSURFACE unnecessary redefinition

humbertodias committed:
SDLL_DisplayFormatAlpha

humbertodias committed:
creating SDLL_JoystickName

humbertodias committed:
stage_screenshot_tool using SDLL_SetVideoMode

humbertodias committed:
SDL_DisplayFormat with out alpha

humbertodias committed:
abstracting SDLL_BlitSurface

humbertodias committed:
wasm: -sTOTAL_MEMORY=128mb

humbertodias committed:
use native SDL_BlitSurface

humbertodias committed:
wasm build working

humbertodias committed:
increasing FPS using SDL_Texture

humbertodias committed:
remove cout to speed up wasm version

humbertodias committed:
replace wasm html template

humbertodias committed:
SDL_PIXELFORMAT_ARGB8888

humbertodias committed:
info wasm localhost

humbertodias committed:
fix: npe of game_screen_scaled

humbertodias committed:
comment out CONFIG+=linux for OSX

humbertodias committed:
remove warn

humbertodias committed:
setAlpha

humbertodias committed:
typo

humbertodias committed:
add SDLL_Quit

humbertodias committed:
fix: game_screen_scaled npe

humbertodias committed:
remove all warnings

humbertodias committed:
Revert "add SDLL_Quit"

This reverts commit 7d62ce5bbaee4e38892e2c8186e32bcfb5faee8e.

humbertodias committed:
add SDLL_ShowVersion

humbertodias committed:
Improve the readability of the separation between SDL1/2

humbertodias committed:
move SDL_Version to main_loop

humbertodias committed:
add SDLL_GetCompiledVersion

humbertodias committed:
fix: emscripten_cancel_main_loop

humbertodias committed:
Add build script for Rockbot with SDL support

This script builds the Rockbot project with optional SDL2 support, checking for dependencies and providing usage instructions.
